### PR TITLE
misc: add klayout

### LINKED ISSUE #175: add klayout package
Upstream: https://github.com/KLayout/klayout
Build recipe: https://github.com/KLayout/klayout#building-without-qt-binding, possibly even without qt with https://github.com/KLayout/klayout/blob/master/build.sh#L93
Dependencies: qt, python, ruby, libcurl, libexpat (but it looks like most of them can be swapped out https://github.com/KLayout/klayout/blob/master/build.sh#L470)

### PR DESCRIPTION
**Pull request by proppy**
Based on @klayoutmatthias's [fork](https://github.com/klayoutmatthias/staged-recipes/tree/klayout/recipes/klayout) of https://github.com/conda-forge/staged-recipes/pull/20396.

Wanted to just build the CLI to aligned with #250 and #193 but couldn't find a build option to disable QT while still keeping `klayout_main`.

Fixes #175

### CODE REVIEW

File: misc/klayout/force-python-dynamic-link.patch [hunk 1]
```diff
@@ -0,0 +1,53 @@
+diff --git a/build.sh b/build.sh
```
Comment (mithro) [inline misc/klayout/force-python-dynamic-link.patch:1]: Log a github issue to get this upstream

File: misc/klayout/build.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+set -e
+set -x
+
+# Identify OS
+UNAME_OUT="$(uname -s)"
+case "${UNAME_OUT}" in
+    Linux*)     OS=Linux;;
+    *)          OS="${UNAME_OUT}"
+                echo "Unknown OS: ${OS}"
+                exit;;
```
Comment (mithro) [inline misc/klayout/build.sh:12]: `exit 1`?

**Review by mithro — APPROVED** [review]
LGTM.

File: misc/klayout/force-python-dynamic-link.patch (hunk 1 above)
Comment (proppy) [reply misc/klayout/force-python-dynamic-link.patch:1]: I suspect @klayoutmatthias link statically intentionally.

**Comment by proppy** [comment]
> LGTM.

Let's not merge before we get the CI passing.

**Comment by proppy** [comment]
@mithro yay it's green! https://github.com/hdl/conda-eda/actions/runs/3751563916/jobs/6372696825 \o/

**Comment by mithro** [comment]
Looks like the CI is still generally broken however?

**Comment by proppy** [comment]
@mithro in my experience the CI is generally flaky for some of the f4pga packages (vtr, yosys-osx) to but I don't see any issues in that runs for the silicon toolchain packages.

**Comment by mithro** [comment]
Merge away then?

**Comment by proppy** [comment]
I suspect @ajelinski will want us to audit the warnings and squash commits first!

**Comment by proppy** [comment]
Identified 4 cluster of warning:
- `WARNING rpath /home/runner/work/conda-eda/conda-eda/workdir/conda-env/conda-bld/klayout_1671649256384/_build_env/lib is outside prefix` which I believe are due to cdt.

- `WARNING Failed to get_static_lib_exports` which according to can safely be ignored https://github.com/hdl/conda-eda/pull/165#issuecomment-1059282395

- `WARNING (klayout,bin/strmclip): Needed DSO lib/libQt5Core.so.5 found in ['qt'] but ['qt'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)` which need fixing.

- `WARNING (klayout): run-exports library package pkgs/main::icu-58.2-he6710b0_3 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)` which need fixing.

**Comment by proppy** [comment]
A few remaining warnings from https://github.com/hdl/conda-eda/actions/runs/3758645638/jobs/6387239942
```
  65630:2022-12-22T15:28:51.4619538Z WARNING (klayout,lib/libklayout_tl.so.0.28.1): Needed DSO lib/libpng16.so.16 found in ['libpng']
  65631:2022-12-22T15:28:51.4620164Z WARNING (klayout,lib/libklayout_tl.so.0.28.1): .. but ['libpng'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
  65632:2022-12-22T15:28:51.4620717Z WARNING (klayout,lib/libklayout_tl.so.0.28.1): Needed DSO lib/libexpat.so.1 found in ['expat']
```
those are cdt so I don't think they are required.
```
  65633:2022-12-22T15:28:51.4621305Z WARNING (klayout,lib/libklayout_tl.so.0.28.1): .. but ['expat'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
  65634:2022-12-22T15:28:51.4621856Z WARNING (klayout,lib/libklayout_tl.so.0.28.1): Needed DSO lib/libz.so.1 found in ['zlib']
  65635:2022-12-22T15:28:51.4622450Z WARNING (klayout,lib/libklayout_tl.so.0.28.1): .. but ['zlib'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
  65717:2022-12-22T15:28:51.4684921Z WARNING 
```
This probably need to be added to `host/run` explicitly.

```
(klayout,lib/klayout/lay_plugins/liblefdef_ui.so.0.28.1): $RPATH/liblefdef.so.0 not found in packages, sysroot(s) nor the missing_dso_whitelist.
  66035:2022-12-22T15:28:51.4879561Z WARNING (klayout,lib/klayout/lay_plugins/libnet_tracer_ui.so.0.28.1): $RPATH/libnet_tracer.so.0 not found in packages, sysroot(s) nor the missing_dso_whitelist.
  66063:2022-12-22T15:28:51.4896117Z WARNING (klayout,lib/klayout/lay_plugins/libpcb_ui.so.0.28.1): $RPATH/libpcb.so.0 not found in packages, sysroot(s) nor the missing_dso_whitelist.
```
assuming those are unused?

**Comment by proppy** [comment]
> This probably need to be added to host/run explicitly.

Done.

**Comment by proppy** [comment]
and rebased.

**Comment by proppy** [comment]
Merging since the build is green and remaining warnings were addressed per https://github.com/hdl/conda-eda/pull/268#issuecomment-1363176244

**Pull request merged**